File: 2) POLIMORFISMO/Ejercicio-5/PYTHON/Celular.py
class Celular:
    def __init__(self, nroTel, dueño, espacio, ram, nroApp):
        self.nroTel = nroTel
        self.dueño = dueño
        self.espacio = espacio
        self.ram = ram
        self.nroApp = nroApp

    def __str__(self):
        return f"Dueño: {self.dueño}, Apps: {self.nroApp}, Espacio: {self.espacio} GB"

    def __iadd__(self, otro):
        self.nroApp += 10
        return self

    def __isub__(self, otro):
        self.espacio -= 5
        if self.espacio < 0:
            self.espacio = 0
        return self

celular = Celular("555-1234", "Estudiante", 100.0, 8, 5)

print("---  Estado Inicial ---")
print(celular)

print("\n--- Aplicando Operador ++ (nroApp + 10) ---")
celular += 1
print(celular)

print("\n--- Aplicando Operador -- (espacio - 5 GB) ---")
celular -= 1
print(celular)

File: 2) POLIMORFISMO/Ejercicio-5/JAVA/Celular.java
public class Celular {
    private String nroTel;
    private String dueño;
    private float espacio;
    private int ram;
    private int nroApp;

    public Celular(String nroTel, String dueño, float espacio, int ram, int nroApp) {
        this.nroTel = nroTel;
        this.dueño = dueño;
        this.espacio = espacio;
        this.ram = ram;
        this.nroApp = nroApp;
    }

    public float getEspacio() {
        return espacio;
    }

    public int getNroApp() {
        return nroApp;
    }

    @Override
    public String toString() {
        return "Dueño: " + dueño + ", Apps: " + nroApp + ", Espacio: " + espacio + " GB";
    }

    // b) Sobrecarga operador ++ (Simulación)
    public void aumentarNroApp() {
        this.nroApp += 10;
    }

    // c) Sobrecarga operador -- (Simulación)
    public void disminuirEspacio() {
        this.espacio -= 5.0f;
        if (this.espacio < 0) {
            this.espacio = 0.0f;
        }
    }
}